amical/_cli/main.py

```diff
@@ -0,0 +1,338 @@
+    )
+    extract_parser.add_argument(
+        "-p",
+        "--plot",
+        action="store_true",
+        help="Plot the diagnostic figures.",
+    )
+    extract_parser.add_argument(
+        "-e",
+        "--expert",
+        action="store_true",
+        help="Save additional plots.",
+    )
+    extract_parser.add_argument(
+        "-f",
+        "--file",
+        default=-1,
+        type=int,
+        help="Select the file index to be extracted (default allows user selection).",
+    )
+    extract_parser.add_argument(
+        "-a",
+        "--all",
+        action="store_true",
+        help="Extract bispectrum from all data files in --datadir.",
+    )
+
+    # __________________________________________________________________________
+    # CALIBRATE STEP
+    # __________________________________________________________________________
+
+    calibrate_parser = subparsers.add_parser(
+        "calibrate",
+        help="Calibrate the extracted NRM data with the associated calibrator.",
+    )
+    # INPUT and OUTPUT directory
+    # __________________________________________________________________________
+
+    calibrate_parser.add_argument(
+        "--datadir",
+        default="extracted/",
+        help="Repository containing the extracted bispectrum (default: %(default)s).",
+    )
+    calibrate_parser.add_argument(
+        "--outdir",
+        default="calibrated/",
+        help="Repository to save the calibrated oifits files (default: %(default)s).",
+    )
+
+    #  ## IMPORTANT PARAMS ##
+    calibrate_parser.add_argument(
+        "--clip",
+        action="store_true",
+        help="Sigma clipping is performed over the calibrator files (if any) to reject "
+        + "bad observables due to seeing conditions, centering, etc (default: %(default)s).",
+    )
+    calibrate_parser.add_argument(
+        "--norm",
+        action="store_true",
+        help="CP uncertaintities are normalized by np.sqrt(n_holes/3.) to not "
+        + "over use the non-independant closure phases (default: %(default)s).",
+    )
+    calibrate_parser.add_argument(
+        "--phscorr",
+        action="store_true",
+        help="Apply a phasor correction due to piston between holes (default: %(default)s).",
+    )
+    calibrate_parser.add_argument(
+        "--atmcorr",
+        action="store_true",
+        help="Apply a atmospheric correction on V2 from seeing and wind shacking issues (default: %(default)s).",
+    )
+
+    # CLI parameters
+    # __________________________________________________________________________
+    calibrate_parser.add_argument(
+        "-p",
+        "--plot",
+        action="store_true",
+        help="Plot the calibrated data.",
+    )
+
+    args = parser.parse_args(argv)
+
+    if args.command == "clean":
+        perform_clean(args)
+    elif args.command == "extract":
+        perform_extract(args)
+    elif args.command == "calibrate":
+        perform_calibrate(args)
+
+    return 0
```

Comment:
So as to make the return value any useful, I suggest refactoring this so that command functions return integers to, and eventually cascades _their_ return value into main as
```suggestion
    retv = 1
    if args.command == "clean":
        retv = perform_clean(args)
    elif args.command == "extract":
        retv = perform_extract(args)
    elif args.command == "calibrate":
        retv = perform_calibrate(args)
    return retv
```